src/taker/api.rs

```diff
@@ -293,6 +304,37 @@ impl Taker {
         })
     }
 
+    /// Helper functions to load and save the offer cache.
+    pub fn load_offer_cache(&self, path: &PathBuf) -> Result<Vec<CachedOffer>, IoError> {
+        if path.exists() {
+            let data = fs::read_to_string(path)?;
+            let offers: Vec<CachedOffer> = serde_json::from_str(&data).map_err(IoError::other)?;
+            Ok(offers)
+        } else {
+            Ok(vec![])
+        }
+    }
+
+    /// Saves the offer cache
+    pub fn save_offer_cache(
+        &self,
+        path: &PathBuf,
+        offers: &Vec<CachedOffer>,
+    ) -> Result<(), IoError> {
+        let data = serde_json::to_string_pretty(offers).map_err(IoError::other)?;
+        fs::write(path, data)
+    }
+
+    #[allow(dead_code)]
+    /// Deletes the offerbook cache file if it exists.
+    pub fn delete_offer_cache(&self) -> Result<(), IoError> {
+        let cache_file = get_taker_dir().join("offerbook_cache.json");
+        if cache_file.exists() {
+            fs::remove_file(cache_file)?;
+        }
+        Ok(())
+    }
```

Comment:
No need for these read/write APIs again. We got OfferBook::write_to_disk() and read_from_disk(). We can modify them instead.

This creates two different serialization of the offerdata, one with Json and one with cbor. Thats redundant and will create problems later. We can just move everything to json.